test/lib/ufe/testMoveCmd.py

```diff
@@ -276,6 +281,118 @@ def ufeSceneItemTranslation(item):
 
         self.runMultiSelectTestMove(ballItems, expected)
 
+    def runTestOpUndo(self, createTransformOp, attrSpec):
+        '''Engine method for op undo testing.'''
+        proxyShape = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+        
+        proxyShapePath = ufe.PathString.path(proxyShape)
+        proxyShapeItem = ufe.Hierarchy.createItem(proxyShapePath)
+        proxyShapeContextOps = ufe.ContextOps.contextOps(proxyShapeItem)
+        proxyShapeContextOps.doOp(['Add New Prim', 'Sphere'])
+
+        spherePath = ufe.PathString.path('%s,/Sphere1' % proxyShape)
+        sphereItem = ufe.Hierarchy.createItem(spherePath)
+
+        spherePrim = mayaUsd.ufe.ufePathToPrim(ufe.PathString.string(spherePath))
+        sphereXformable = UsdGeom.Xformable(spherePrim)
+        createTransformOp(self, sphereXformable)
+
+        # Set the edit target to the session layer, then move.
+        stage = mayaUsd.ufe.getStage(proxyShape)
+        sessionLayer = stage.GetLayerStack()[0]
+        stage.SetEditTarget(sessionLayer)
+
+        sn = ufe.GlobalSelection.get()
+        sn.clear()
+        sn.append(sphereItem)
+
+        cmds.move(0, 10, 0, relative=True, os=True, wd=True)
+
+        sphereT3d = ufe.Transform3d.transform3d(sphereItem)
+        assertVectorAlmostEqual(self, sphereT3d.translation().vector, [0, 10, 0])
+
+        primSpec = stage.GetEditTarget().GetPrimSpecForScenePath('/Sphere1')
+
+        # Writing to the session layer has created a primSpec in that layer,
+        # with the xformOp:transform attrSpec.
+        self.assertIsNotNone(primSpec)
+        self.assertIn(attrSpec, primSpec.attributes)
+
+        # Undo: primSpec in session layer should be gone.
+        cmds.undo()
+
+        assertVectorAlmostEqual(self, sphereT3d.translation().vector, [0, 0, 0])
+
+        primSpec = stage.GetEditTarget().GetPrimSpecForScenePath('/Sphere1')
+        self.assertIsNone(primSpec)
+
+        cmds.redo()
+
+        assertVectorAlmostEqual(self, sphereT3d.translation().vector, [0, 10, 0])
+
+        # Undo after redo must also remove primSpec.
+        cmds.undo()
+
+        assertVectorAlmostEqual(self, sphereT3d.translation().vector, [0, 0, 0])
+
+        primSpec = stage.GetEditTarget().GetPrimSpecForScenePath('/Sphere1')
+        self.assertIsNone(primSpec)
```

Comment:
When writing to the session layer, check for the presence / absence of the primSpec.